app/gui/controller/double-representation/src/context_switch.rs

```diff
@@ -150,6 +151,18 @@ impl ContextSwitchExpression {
         let args = vec![context, environment];
         ast::prefix::Chain::new(func, args).into_ast()
     }
+
+    /// Remove the context switch expression from the given AST. Does nothing if `ast` does not
+    /// contain any context switch expression.
+    pub fn remove_from_ast(ast: &Ast) -> Ast {
+        if ContextSwitchExpression::parse(ast).is_some() {
+            let crumb = ast::crumbs::InfixCrumb::RightOperand.into();
+            let rarg = ast.get(&crumb).unwrap_or(ast);
+            rarg.clone()
+        } else {
+            ast.clone()
+        }
+    }
```

Comment:
Name and docs implies some modification, but the original AST is not modified, actually. For macros, we have `without_macros` function - maybe here should be a similar name?

Reply:
Renamed